src/pybind/mgr/cephadm/inventory.py

```diff
@@ -291,13 +301,30 @@ def update_host_daemons(self, host, dm):
     def update_host_facts(self, host, facts):
         # type: (str, Dict[str, Dict[str, Any]]) -> None
         self.facts[host] = facts
-        self.last_facts_update[host] = datetime.datetime.utcnow()
+        self.last_facts_update[host] = datetime_now()
+
+    def devices_changed(self, host: str, b: List[inventory.Device]) -> bool:
+        a = self.devices[host]
+        if len(a) != len(b):
+            return True
+        aj = {d.path: d.to_json() for d in a}
+        bj = {d.path: d.to_json() for d in b}
+        if aj != bj:
+            self.mgr.log.info("Detected new or changed devices on %s" % host)
+            return True
+        return False
```

Comment:
looks similar to 

https://github.com/ceph/ceph/blob/faa93b751dc13003b23370f769a8ea252972c3dc/src/python-common/ceph/deployment/inventory.py#L18-L19

Sadly this is in the somewhat unused `Devices` class.